### ISSUE TITLE
How to evaluate the training process?

### ISSUE THREAD

**Issue by leyuan-sun** [JiawangBian/SC-SfMLearner-Release#76]
I am a noob at DL, I successfully use my own data to do training right now. Could u plz tell me how to evaluate my training is successful or not (from some visualization or metrics)? Thanks ahead!

**Comment by JiawangBian** [comment]
You can see the validation error (AbsRel). if it is around or lower than 0.2, it will be good. You can also see the validation accuracy (a1). It means what percentage points are correct. Besides, you can find the depth visualization.

**Comment by leyuan-sun** [comment]
In tensorboard, I have Consistency_loss(increasing), Photo_loss(Fluctuating up and down, gradually decreasing), Smooth_loss(increasing), Total_loss(increasing), disparity_smooth_loss(increasing), geometry_consistency_loss(Fluctuating up and down) , photometric_loss(decreasing),total_loss(decreasing, since disparity_smooth_loss and gemoetry_consistency_loss is very samll comaperd with photometric_loss).

1. what is the difference between the former 4 tpyes' loss with the latter 4?
2. where I can chek AbsRel, and a1? I did not find them in tensorboard, I use the resnet50pose256 scripts to train my own data.
3. In tensorboard Images, what is the difference between val_Depth_Output and val_Dispnet_Output_Normlized? From, visualization, normlized one is looks good, but val_depth_out is all red.

**Comment by JiawangBian** [comment]
1. First 4 are validation loss. The latter 4 are training loss.
2. if you do not have GT depth. you cannot get them.
3. val_Dispnet_Output_Normlized shows the disparity (inverse depth). val_depth_out shows the depth. You can just see the normalized one.

**Comment by leyuan-sun** [comment]
感谢回复，我还是用中文吧，那其中有些loss在一直增加该怎么办，尽管total loss是在减少的，但是那是因为photometric loss在减少并且它的数量级远远大于另外的。

**Comment by JiawangBian** [comment]
能收敛到一个稳定的范围就行。

**Comment by leyuan-sun** [comment]
增加到一定然后再收敛也可以么，我在linkedin加你了，希望你能通过哈。谢谢！

**Comment by leyuan-sun** [comment]
Sorry, one more question, how can I get the mask image during training and inference?

**Comment by JiawangBian** [comment]
可以在loss.py里找到 weight mask。你需要写几行代码手动保存。

**Comment by leyuan-sun** [comment]
Right now, I can visualize weight_mask during training, however, the inference process does not use loss_function part, how can I obtain them during testing? Thanks!

**Comment by JiawangBian** [comment]
see 'validate_without_gt' function in train.py

**Comment by leyuan-sun** [comment]
恩，我已经可以在训练的时候看到weight_mask了。 我想问怎么在evaluation的时候看到这个mask？

**Comment by leyuan-sun** [comment]
Anyway, thanks. I can visualize the weight_mask with my own training data in the training process. 
![2021-04-20 11:29:20 735805](https://user-images.githubusercontent.com/41459267/115346629-60e5ee00-a1eb-11eb-8468-0151deaaf7d1.jpg)
